frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -1074,11 +1105,11 @@ static Type::Genericity getFieldsGenericity(Context* context,
   }
 
   if (context->isQueryRunning(fieldsForTypeDeclQuery,
-                              std::make_tuple(ct, DefaultsPolicy::IGNORE_DEFAULTS)) ||
+                              std::make_tuple(ct, DefaultsPolicy::IGNORE_DEFAULTS, false)) ||
```

Comment:
Would you be up for adjusting `getFieldsGenericity` here to use `isFieldSyntacticallyGeneric` instead of even trying to fully resolve the fields? This logic about "is query running" is very not ideal.